code/modules/clothing/clothing.dm

```diff
@@ -699,6 +699,26 @@ BLIND     // can't see anything
 
 	. = ..()
 
+/obj/item/clothing/under/serialize()
+	var/data = ..()
+	var/list/accessories_list = list()
+	data["accessories"] = accessories_list
+	for(var/obj/item/clothing/accessory/A in accessories)
+		accessories_list.len++
+		accessories_list[accessories_list.len] = A.serialize()
```

Comment:
Is there a reason for increasing the list's length and making it associative like this?

I would say just do `accessories_list += A.serialize()`

Reply:
I cribbed that from https://github.com/ParadiseSS13/Paradise/blob/master/code/game/objects/items/weapons/storage/storage.dm#L685.

I believe it has to do with the fact that the result of A.serialize() is a list, but if you attempt the `+=` operation, the resultant JSON is malformed; instead of separate nested objects for each accessories, there's just key-value pairs of the last accessory's type repeatedly:

```
        ...
        {
            "type": "/obj/item/clothing/under/color/green",
            "accessories": {
                "type": "/obj/item/clothing/accessory/armband/engine",
                "type": "/obj/item/clothing/accessory/armband/engine",
                "type": "/obj/item/clothing/accessory/armband/engine"
            }
        },
        ...
```

The comment above the implementation I cribbed says "I wish I could just insert a list instead of it reading it the wrong way" so I assume this is what they're referring to, that the `+=` idiom doesn't work as expected in this situation.

Reply:
That's really strange that it would duplicate the same accessory more than once, since it's just a single loop, but alright 👍